engine/language-server/src/test/scala/org/enso/languageserver/libraries/ComponentGroupsValidatorSpec.scala

```diff
@@ -29,20 +29,17 @@ class ComponentGroupsValidatorSpec extends AnyWordSpec with Matchers {
           "Error message"
         )
       )
-
-      validator.validate(testPackages) shouldEqual testPackages.map { config =>
-        config.componentGroups match {
-          case Right(_) =>
-            Right(config)
-          case Left(error) =>
-            Left(
-              ValidationError.InvalidComponentGroups(
-                libraryName(config),
-                error.getMessage()
-              )
-            )
+      val expected = testPackages.map { config =>
+        val groups = config.componentGroups.left.map { error =>
+          ValidationError.InvalidComponentGroups(
+            libraryName(config),
+            error.getMessage()
+          )
```

Comment:
This test is re-using a big chunk of the logic that it is testing - so quite likely the result will be consistent.

But wouldn't it be better to just manually type out the expected errors here?

Then a bug inside of the internal logic will be detectable (currently both sides may do the same wrong thing and tests won't notice).
Plus the test will serve as a clear example of what kind of errors we expect - which is great documentation.

Reply:
The `error` was defined in the test (in the definition of `testPackages`). So, here we transform our test input into the expected output, and not the validation output into the expected output. I think it should be fine to reuse the test input here